### PR TITLE
[Port] Refactors machine_stat and is_processing() to process on demand + Changes obj_break on machines to use parent calls

### PR DESCRIPTION
**Pull request by PestoVerde322**
## About The Pull Request

This PR aims at modernizing the code by porting https://github.com/tgstation/tgstation/pull/46485 and https://github.com/tgstation/tgstation/pull/53150/, which are going to be rather useful for future projects (see for example https://github.com/tgstation/tgstation/pull/66059)

46485 reworks the way stat |= BROKEN was done was a mess, this makes everywhere use obj_break with proper parent calls and adds a signal for when a machine enters the broken state.

53150 aims at handling changes in the variable through set_machine_stat() so we can react to the event of it changing.
Offers no functional changes here as is, but should allow things that depend on it to process on-demand.

## Why It's Good For The Game

Refactoring and modernization of the code are quite in demand, also increases code quality.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

It works in game... do i need to post a screen?

</details>

## Changelog
:cl:
add: Added new way on how broken machinery works
code: added a new signal for broken machinery
refactor: refactored machine_stat and is_processing
refactor: refactored relay.dm with a better logic for it's operational logic
/:cl:

### CODE REVIEW

File: code/modules/NTNet/relays.dm [hunk 1]
```diff
@@ -24,25 +27,46 @@
 	var/dos_capacity = 500		// Amount of DoS "packets" in buffer required to crash the relay
 	var/dos_dissipate = 0.5		// Amount of DoS "packets" dissipated over time.
 
+///Proc called to change the value of the `relay_enabled` variable and append behavior related to its change.
+/obj/machinery/ntnet_relay/proc/set_relay_enabled(new_value)
+	if(new_value == relay_enabled)
```
Comment (itsmeow) [inline code/modules/NTNet/relays.dm:32]: sort of undocumented change

File: code/modules/power/floodlight.dm [hunk 2]
```diff
@@ -106,12 +106,9 @@
 	..()
 
 /obj/machinery/power/floodlight/obj_break(damage_flag)
-	if(!(flags_1 & NODECONSTRUCT_1))
-		playsound(loc, 'sound/effects/glassbr3.ogg', 100, 1)
-		var/obj/structure/floodlight_frame/F = new(loc)
```
Comment (itsmeow) [inline code/modules/power/floodlight.dm:111]: where did all this go? does the parent handle it?

**Comment by itsmeow** [comment]
Probably doesn't need a TM

File: code/modules/power/floodlight.dm (hunk 2 above)
Comment (PestoVerde322) [reply code/modules/power/floodlight.dm:111]: Botched the port, fixed now

File: code/modules/NTNet/relays.dm (hunk 1 above)
Comment (PestoVerde322) [reply code/modules/NTNet/relays.dm:32]: Added in the 🆑, closing

**Comment by itsmeow** [comment]
Don't include anecdotes in CL

**Comment by itsmeow** [comment]
I am not convinced of the usefulness of renaming `stat` to `machine_stat`.

Maybe there should be an `is_broken` proc, that returns `machine_stat & BROKEN` instead. Although, I'm not sure of the performance implications of this, from a code reability standpoint it's much better.

**Comment by PowerfulBacon** [comment]
Having it not named stat is nice, since Ctrl-F is easier

**Review by PowerfulBacon — DISMISSED** [review]

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**